disnake/ext/commands/interaction_bot_base.py

```diff
@@ -139,6 +139,90 @@ def _format_diff(diff: _Diff) -> str:
     return "\n".join(f"| {line}" for line in lines)
 
 
+def _parse_sync_warning(exception: Exception, commands: List[ApplicationCommand]) -> str:
+    if (
+        not isinstance(exception, HTTPException)
+        or exception.status != 400
+        or not isinstance(exception._errors, dict)
+    ):
+        return str(exception)
+
+    try:
+        sync_warnings: List[str] = []
+        for command_num, error in exception._errors.items():
+            command = commands[int(command_num)]
+
+            message = f"In {command.name}:\n"
+            for key, value in error.items():
+                message = _parse_options(
+                    message,
+                    key,
+                    value,
+                    command.options if isinstance(command, SlashCommand) else [],
+                    2,
+                    "Option",
+                )
+
+            sync_warnings.append(message)
+
+        return "\n".join(sync_warnings)
+    except Exception:
+        return str(exception)
+
+
+def _parse_options(
+    message: str,
+    key: str,
+    value: Any,
+    options: Union[List[Option], List[OptionChoice]],
+    indent_level: int,
+    metadata: str,
+):
+
+    # When we are parsing an option or choice, the metadata param will tell us where the `options` came from
+    # This metadata is attached to the message to show if the key is an Option or Choice
+    # We use new_metadata for when we parse a non-decimal key, since we need to carry this metadata
+    # over to the next iteration.
+
+    # It is helpful to know the error dict looks something like
+    # {
+    #     "options": {
+    #         "0": {
+    #             "choices": {
+    #                 "0": {
+    #                     "name": {"_errors": ...}
+    #                 }
+    #             }
+    #         }
+    #     }
+    # }
+
+    new_metadata = ""
+    if key.isdecimal():
+        new_metadata = f" ({metadata} {key})"
```

Comment:
Could consider making these 1-indexed instead of 0-indexed. Thoughts?

Reply:
Option 0 is fine